From 93beb0278e806faeb977a87a2eb1dbab31de3ae3 Mon Sep 17 00:00:00 2001
From: Matthew Gerlach <mgerlach@altera.com>
Date: Fri, 5 Apr 2013 09:10:26 -0700
Subject: [PATCH 012/172] FogBugz #103239: Intermittent loss of ethernet
 transmission.

This bug does not really have anything to do with ethernet.
The problem was a memory coherency problem between the cpu
and a dma engine, in this case the ethernet mac.  In short,
while dma writes by the ethernet were not getting cached in L2,
they were getting cached in L1.  The good news is a single bit
change to the programming of the PL310 fixes the problem.

Signed-off-by: Matthew Gerlach <mgerlach@altera.com>
[czou:Original patch taken from
https://github.com/altera-opensource/linux-socfpga.git socfpga-4.1]
Signed-off-by: czou <cao.zou@windriver.com>
---
 arch/arm/mach-socfpga/socfpga.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/arch/arm/mach-socfpga/socfpga.c b/arch/arm/mach-socfpga/socfpga.c
index 2963760..49537b0 100644
--- a/arch/arm/mach-socfpga/socfpga.c
+++ b/arch/arm/mach-socfpga/socfpga.c
@@ -172,7 +172,8 @@ static const char *altera_dt_match[] = {
 
 DT_MACHINE_START(SOCFPGA, "Altera SOCFPGA")
 	.l2c_aux_val	= L310_AUX_CTRL_DATA_PREFETCH |
-			L310_AUX_CTRL_INSTR_PREFETCH,
+			L310_AUX_CTRL_INSTR_PREFETCH |
+			L2C_AUX_CTRL_SHARED_OVERRIDE,
 	.l2c_aux_mask	= ~0,
 	.smp		= smp_ops(socfpga_smp_ops),
 	.map_io		= socfpga_map_io,
-- 
1.9.1

